ShipEngine/ShipEngineClient.cs

```diff
@@ -21,28 +22,30 @@ public class ShipEngineClient
         /// Options for serializing the method call params to JSON.
         /// A separate inline setting is used for deserializing the response
         /// </summary>
-        protected readonly JsonSerializerOptions JsonSerializerOptions;
+        protected static readonly JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions
+        {
+            DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+            PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower,
+            PropertyNameCaseInsensitive = true,
+            WriteIndented = true,
+            Converters = { new JsonStringEnumMemberConverter() }
+        };
+
+        private const string JsonMediaType = "application/json";
 
         /// <summary>
-        /// Constructor for ShipEngineClient
+        /// Modifies the client request before it is sent
         /// </summary>
-        public ShipEngineClient()
-        {
-            JsonSerializerOptions = new JsonSerializerOptions
-            {
-                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
-                PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower,
-                PropertyNameCaseInsensitive = true,
-                WriteIndented = true,
-                Converters = { new JsonStringEnumMemberConverter() }
-            };
-        }
+        protected Action<HttpRequestMessage>? ModifyRequest { get; set; }
```

Comment:
Does this need to be protected?  We want users to be able to set it. I know we talked about a `.WithModifiedRequest()` method that would return a new instance with this field set, but is there any reason to force them to use that method? Seems like it should be fine to let people do `client.ModifyRequest = r => ...` as well

Reply:
fair question - I think likely not, but am just going to remove the modify request bit from this effort while that gets further consideration, so as not to cause more delay with the rest of the functionality I'm looking to deliver here